### PR TITLE
Connect to server

### PR DESCRIPTION
**Pull request by Darmody**

### CODE REVIEW

**Comment by Darmody** [comment]
@calebx you should review this one first

File: src/sagas/videos/fetch.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  return yield call(apiClient, {
+    endpoint: 'videos/seeds/',
+  })
+}
+
+export function* fetchRequest() {
+  yield httpRequestHandler(
+    executeFetchRequest,
+    null,
+    [
```
Comment (calebx) [inline src/sagas/videos/fetch.js]: {} instead of null?

Comment (Darmody) [inline src/sagas/videos/fetch.js]: No, `null` means no params.

File: src/sagas/intercepters/httpRequestHandler.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import requestHandler from './requestHandler'
+import successHandler from './successHandler'
+import failureHandler from './failureHandler'
+
+function* httpRequestHandler(request, params, types, schema, meta) {
+  let response = {}
+  try {
+    yield requestHandler()
+    response = yield request(params)
+
```
Comment (calebx) [inline src/sagas/intercepters/httpRequestHandler.js]: how about:
`
const [_ , goodType, badType] = types
...
successHandler(response, goodType, schema, meta)
failureHandler(response, badType, meta)
`
give up [1] [2] would be nicer

Comment (Darmody) [inline src/sagas/intercepters/httpRequestHandler.js]: right

File: src/httpClients/defaultClient.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      'Content-Type': 'application/json',
+    }
+
+    theBody = JSON.stringify(body)
+  }
+
+  return fetch(`${endpoint}?${qs.stringify(queryString)}`, {
+    method,
+    headers: theHeaders,
+    body: method === 'GET' ? undefined : theBody,
```
Comment (calebx) [inline src/httpClients/defaultClient.js]: move to ⬆️ codes?
here could be `body: theBody`

File: src/components/VideoList/index.js [hunk 4]
```diff
@@ -1,114 +1,47 @@
...
-]
-
-const VideoList = () => (
+const VideoList = ({ seeds }) => (
   <div styleName="video-list">
     {
-      fakeVideoList.map((video, index) => (
-        <VideoItem key={index} video={video} />
+      _.toPairs(seeds).map(([channel, videos], index) => (
+        <div styleName="channel-section" key={index}>
```
Comment (calebx) [inline src/components/VideoList/index.js]: 👍

File: src/httpClients/defaultClient.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  credentials = 'no-cors',
+}) => {
+  let theHeaders = headers
+  let theBody = body
+
+  if (json) {
+    theHeaders = {
+      ...headers,
+      'Content-Type': 'application/json',
+    }
```
Comment (calebx) [inline src/httpClients/defaultClient.js]: could here be simply:
```
if (json) theHeaders['Content-Type'] = 'application/json'
```

File: src/httpClients/defaultClient.js (hunk 3 above)
Comment (calebx) [inline src/httpClients/defaultClient.js]: move to ⬆️ codes?
here could be `body: theBody`

File: src/httpClients/defaultClient.js (hunk 5 above)
Comment (calebx) [inline src/httpClients/defaultClient.js]: could here be simply:
```
if (json) theHeaders['Content-Type'] = 'application/json'
```

File: src/httpClients/defaultClient.js (hunk 3 above)
Comment (Darmody) [inline src/httpClients/defaultClient.js]: Agree

File: src/httpClients/defaultClient.js (hunk 5 above)
Comment (Darmody) [inline src/httpClients/defaultClient.js]: Yeah, though too much

**Pull request closed without merging**